deploy-aks/azext_aks_deploy/dev/functionapp/up.py

```diff
@@ -0,0 +1,86 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.prompting import prompt
+from knack.log import get_logger
+from knack.util import CLIError
+
+from azext_aks_deploy.dev.common.git import get_repository_url_from_local_repo, uri_parse
+from azext_aks_deploy.dev.common.github_api_helper import (Files, get_work_flow_check_runID, get_languages_for_repo,
+                                                           push_files_github, get_check_run_status_and_conclusion,
+                                                           get_github_pat_token)
+from azext_aks_deploy.dev.common.github_workflow_helper import poll_workflow_status
+from azext_aks_deploy.dev.common.github_azure_secrets import get_azure_credentials
+from azext_aks_deploy.dev.common.utils import get_repo_name_from_repo_url
+from azext_aks_deploy.dev.common.const import ( APP_NAME_DEFAULT, APP_NAME_PLACEHOLDER)
+
+logger = get_logger(__name__)
+
+def functionapp_deploy(repository=None, skip_secrets_generation=False, no_wait=False):
+    """Build and Deploy to Azure FunctionApp via GitHub actions
+    :param repository: GitHub repository URL e.g. https://github.com/azure/azure-cli.
+    :type repository: str
+    :param skip_secrets_generation : Flag to skip generating Azure credentials.
+    :type skip_secrets_generation: bool
+    :param no_wait : Do not wait for workflow completion.
+    :type no_wait bool
```

Comment:
Does this work?
no_wait is a built-in azure cli parameter, which i believe only works for create/update operations